cmd/tracee-ebpf/main.go

```diff
@@ -175,30 +172,18 @@ func main() {
 
 			// OS kconfig information
 
-			kernelConfig, err := helpers.InitKernelConfig()
-			if err == nil { // do not fail (yet ?) if we cannot init kconfig
-				kernelConfig.AddNeeded(helpers.CONFIG_BPF, helpers.BUILTIN)
-				kernelConfig.AddNeeded(helpers.CONFIG_BPF_SYSCALL, helpers.BUILTIN)
-				kernelConfig.AddNeeded(helpers.CONFIG_KPROBE_EVENTS, helpers.BUILTIN)
-				kernelConfig.AddNeeded(helpers.CONFIG_BPF_EVENTS, helpers.BUILTIN)
-				missing := kernelConfig.CheckMissing() // do fail if we found os-release file and it is not enough
-				if len(missing) > 0 {
-					return fmt.Errorf("missing kernel configuration options: %s", missing)
-				}
-			} else {
-				if debug {
-					fmt.Fprintf(os.Stderr, "KConfig: warning: could not check enabled kconfig features\n(%v)\n", err)
-					fmt.Fprintf(os.Stderr, "KConfig: warning: assuming kconfig values, might have unexpected behavior\n")
-				}
+			kernelConfig, err := initialize.KernelConfig()
+			if err != nil {
+				return err
 			}
 
 			// decide BTF & BPF files to use based on kconfig, release & environment
-			err = prepareBpfObject(&cfg, kernelConfig, OSInfo)
+			err = initialize.BpfObject(&cfg, kernelConfig, OSInfo, traceeInstallPath, version)
```

Comment:
Not directly related to your PR, but as we pass kernelConfig here, which might not be initialized, we can get an error while trying to access it later down the road.

Reply:
True, I will open a separate issue as a potential bug for that, since this behavior exists before as well.